Can't build for net46 (net451, net452 works fine)

## Steps to reproduce
Use the https://github.com/aspnet/cli-samples/tree/master/HelloMvc sample. 
Add net46 as the target framework
Restore (restores fine)
Build (`dotnet build -f net46`)
## Expected  behavior
Project builds successfully

## Actual behavior
```
PS C:\Source\cli-samples\HelloMvc> dotnet build -f net46
Project HelloMvc (.NETFramework,Version=v4.6) will be compiled because expected outputs are missing
Compiling HelloMvc for .NETFramework,Version=v4.6
C:\Source\cli-samples\HelloMvc\project.json(15,42): error NU1001: The dependency System.Diagnostics.FileVersionInfo >= 4.0.0-rc2-24027 could not be resolved.
C:\Source\cli-samples\HelloMvc\project.json(15,42): error NU1001: The dependency System.Diagnostics.StackTrace >= 4.0.1-rc2-24027 could not be resolved.
C:\Source\cli-samples\HelloMvc\project.json(15,42): error NU1001: The dependency System.Text.Encoding.CodePages >= 4.0.1-rc2-24027 could not be resolved.
C:\Source\cli-samples\HelloMvc\project.json(15,42): error NU1001: The dependency System.Threading.Thread >= 4.0.0-rc2-24027 could not be resolved.
C:\Source\cli-samples\HelloMvc\project.json(15,42): error NU1001: The dependency System.Xml.XmlDocument >= 4.0.1-rc2-24027 could not be resolved.
C:\Source\cli-samples\HelloMvc\project.json(15,42): error NU1001: The dependency System.Xml.XPath.XDocument >= 4.0.1-rc2-24027 could not be resolved.

Compilation failed.
    0 Warning(s)
    6 Error(s)

Time elapsed 00:00:00.0945403
```

Note when targeting net451 or net452 the project builds just fine. 

## Environment data
`dotnet --info` output:

```
PS C:\Source\cli-samples\HelloMvc> dotnet --info
.NET Command Line Tools (1.0.0-rc2-002670)

Product Information:
 Version:     1.0.0-rc2-002670
 Commit Sha:  f0018642a9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
```

--- a/TestAssets/TestProjects/AppWithNet46AndRoslyn/.noautobuild
+++ b/TestAssets/TestProjects/AppWithNet46AndRoslyn/.noautobuild
@@ -1,0 +1,1 @@
+noautobuild

--- a/TestAssets/TestProjects/AppWithNet46AndRoslyn/Program.cs
+++ b/TestAssets/TestProjects/AppWithNet46AndRoslyn/Program.cs
@@ -1,0 +1,10 @@
+namespace AppWithNet46AndRoslyn
+{
+    public class MyApp
+    {
+        public static void Main()
+        {
+            System.Console.WriteLine("Hello, World!");
+        }
+    }
+}

--- a/TestAssets/TestProjects/AppWithNet46AndRoslyn/project.json
+++ b/TestAssets/TestProjects/AppWithNet46AndRoslyn/project.json
@@ -1,0 +1,11 @@
+{
+  "buildOptions": {
+    "emitEntryPoint": true
+  },
+  "dependencies": {
+    "Microsoft.CodeAnalysis.CSharp": "1.3.0-beta1-20160429-01"
+  },
+  "frameworks": {
+    "net46": { }
+  }
+}

--- a/src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs
+++ b/src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs
@@ -440,15 +440,21 @@ namespace Microsoft.DotNet.ProjectModel
                     package.HasCompileTimePlaceholder &&
                     !TargetFramework.IsPackageBased)
                 {
+                    // requiresFrameworkAssemblies is true whenever we find a CompileTimePlaceholder in a non-package based framework, even if
+                    // the reference is unresolved. This ensures the best error experience when someone is building on a machine without
+                    // the target framework installed.
+                    requiresFrameworkAssemblies = true;
+
                     var newKey = new LibraryKey(library.Identity.Name, LibraryType.ReferenceAssembly);
                     var dependency = new LibraryRange(library.Identity.Name, LibraryType.ReferenceAssembly);
 
-                    // If the framework assembly can't be resolved then mark it as unresolved but still replace the package
-                    // dependency
-                    var replacement = referenceAssemblyDependencyResolver.GetDescription(dependency, TargetFramework) ??
-                                        UnresolvedDependencyProvider.GetDescription(dependency, TargetFramework);
-
-                    requiresFrameworkAssemblies = true;
+                    var replacement = referenceAssemblyDependencyResolver.GetDescription(dependency, TargetFramework);
+                    
+                    // If the reference is unresolved, just skip it.  Don't replace the package dependency
+                    if (replacement == null)
+                    {
+                        continue;
+                    }
 
                     // Remove the original package reference
                     libraries.Remove(pair.Key);

--- a/test/dotnet-build.Tests/GivenDotnetBuildBuildsProjects.cs
+++ b/test/dotnet-build.Tests/GivenDotnetBuildBuildsProjects.cs
@@ -100,5 +100,20 @@ namespace Microsoft.DotNet.Tools.Builder.Tests
                 "TestLibraryWithXmlDoc.xml"
             });
         }
+
+        [WindowsOnlyFact]
+        public void It_builds_projects_targeting_net46_and_Roslyn()
+        {
+            var testInstance = TestAssetsManager
+                .CreateTestInstance("AppWithNet46AndRoslyn")
+                .WithLockFiles();
+
+            var testProject = Path.Combine(testInstance.TestRoot, "project.json");
+
+            new BuildCommand(testProject)
+                .Execute()
+                .Should()
+                .Pass();
+        }
     }
 }

